generated/nixnetsocket/nixnetsocket.proto

```diff
@@ -83,6 +84,22 @@ message SockOptData {
   }
 }
 
+message AddrInfo {
+  int32 flags = 1;
+  int32 family = 2;
+  int32 sock_type = 3;
+  int32 protocol = 4;
+  SockAddr addr = 5;
+  string canon_name = 6;
+}
+
+message AddrInfoHint {
```

Comment:
I separated out a separate message for the `hints` argument. I saw comments in several spots that only the flags, family, socktype, and protocol should be set for it (the rest 0 or NULL) so I thought it'd be simpler to have this separate type for the client to use with its own input converter with the remaining fields nulled out.